src/contentsharecontroller/DefaultContentShareController.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import AudioVideoController from '../audiovideocontroller/AudioVideoController';
+import MeetingSessionConfiguration from '../meetingsession/MeetingSessionConfiguration';
+import MeetingSessionCredentials from '../meetingsession/MeetingSessionCredentials';
+import ContentShareConstants from './ContentShareConstants';
+import ContentShareController from './ContentShareController';
+import ContentShareMediaStreamBroker from './ContentShareMediaStreamBroker';
+
+export default class DefaultContentShareController implements ContentShareController {
+  static createContentShareMeetingSessionConfigure(
+    configuration: MeetingSessionConfiguration
+  ): MeetingSessionConfiguration {
+    let contentShareConfiguration = new MeetingSessionConfiguration();
+    contentShareConfiguration.meetingId = configuration.meetingId;
+    contentShareConfiguration.urls = configuration.urls;
+    contentShareConfiguration.credentials = new MeetingSessionCredentials();
+    contentShareConfiguration.credentials.attendeeId =
+      configuration.credentials.attendeeId + ContentShareConstants.Modality;
+    contentShareConfiguration.credentials.joinToken =
+      configuration.credentials.joinToken + ContentShareConstants.Modality;
+    return contentShareConfiguration;
+  }
+
+  constructor(
+    private mediaStreamBroker: ContentShareMediaStreamBroker,
+    private audioVideo: AudioVideoController
+  ) {}
+
+  async startContentShare(stream: MediaStream): Promise<void> {
+    if (!stream) {
+      return;
+    }
+    this.mediaStreamBroker.mediaStream = stream;
+    this.audioVideo.start();
+    if (this.mediaStreamBroker.mediaStream.getVideoTracks().length > 0) {
+      this.audioVideo.videoTileController.startLocalVideoTile();
+    }
+  }
+
+  async startContentShareFromScreenCapture(sourceId?: string): Promise<void> {
+    let mediaStream = await this.mediaStreamBroker.acquireScreenCaptureDisplayInputStream(sourceId);
+    this.startContentShare(mediaStream);
+  }
+
+  pauseContentShare(): void {
+    this.mediaStreamBroker.toggleMediaStream(false);
+  }
+
+  unpauseContentShare(): void {
+    this.mediaStreamBroker.toggleMediaStream(true);
+  }
+
+  stopContentShare(): void {
+    this.audioVideo.stop();
+    this.mediaStreamBroker.cleanup();
```

Comment:
```this.audioVideo.stop()``` is a non-async call, but it performs the async "actionDisconnect" operation. Could you confirm that calling ```cleanup``` while running "actionDisconnect" has no conflict?

Reply:
I think this is fine. There is a question as to who should be responsible for stopping the media tracks, the API consumer or the API implementation. I think what we may want to do later is provide an optional boolean flag that allows for running cleanup or not, but we can leave this as is for now since it's reasonable to clean up a MediaStream that is no longer being shared.